tests/test_schema.py

```diff
@@ -683,6 +683,65 @@ def test_deserializing_relationship_fields(self):
         assert data['author'] == "1"
         assert data['comments'] == ["1"]
 
+    def test_deserializing_nested_relationship_fields(self):
+        class RelationshipWithSchemaCommentSchema(Schema):
+            id = fields.Str()
+            body = fields.Str(required=True)
+            author = fields.Relationship(
+                schema=AuthorSchema, many=False, type_='people'
+            )
+
+            class Meta:
+                type_ = 'comments'
+                strict = True
+
+        class RelationshipWithSchemaArticleSchema(Schema):
+            id = fields.Integer()
+            body = fields.String()
+            comments = fields.Relationship(
+                schema=RelationshipWithSchemaCommentSchema, many=True, type_='comments'
+            )
+
+            class Meta:
+                type_ = 'articles'
+                strict = True
+
+        article = self.article.copy()
+        article['included'] = [
+            {
+                "id": "1",
+                "type": "comments",
+                "attributes": {
+                    "body": "Test comment"
+                },
+                "relationships": {
+                    "author": {
+                        'data': {
+                            "type": "people",
+                            "id": "2"
+                        }
+                    }
+                }
+            },
+            {
+                'id': "2",
+                'type': "people",
+                'attributes': {
+                    "first_name": "Marshmallow Jr",
+                    "last_name": "JsonAPI"
+                }
+            }
+        ]
+
+        included_author = filter(lambda item: item['type'] == 'people', article['included'])
+        included_author = list(included_author)[0]
```

Comment:
Minor: I believe this can be simplified to 

```python
included_author = next(filter(lambda item: item['type'] == 'people', article['included']))
```